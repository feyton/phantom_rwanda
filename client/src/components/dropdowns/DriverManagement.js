import React from 'react';
import { toast } from 'react-toastify';

const handleEdit = () => {
  toast('Edit clicked');
};
const handleDelete = () => {
  toast('Delete clicked');
};
const handlePermission = () => {
  toast('Permissions clicked');
};

function DriverManagement() {
  const handleChange = (e) => {
    switch (e.target.value) {
      case 'edit':
        return handleEdit();
      case 'delete':
        return handleDelete();
      case 'permission':
        return handlePermission();
    }
  };
  return (
    <div className="z-10">
      <select
        onChange={handleChange}
        role="select"
        name=""
        id=""
        className="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
      >
        <option hidden>Manage</option>
        <option className="cursor-pointer" value="edit">
          Edit
        </option>
        <option className="cursor-pointer" value="delete">
          Delete
        </option>
        <option className="cursor-pointer" value="permission">
          Permissions
        </option>
      </select>
    </div>
  );
}

export default DriverManagement;
