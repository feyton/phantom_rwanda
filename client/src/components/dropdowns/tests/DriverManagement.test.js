import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import DriverManagement from '../DriverManagement.js';

describe('Bus Management test', () => {
  it('Should render drivers management page', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <DriverManagement />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('should edit a bus', async () => {
    const handleEditMock = jest.fn();
    const elem = render(
      <BrowserRouter>
        <DriverManagement handleEdit={handleEditMock} />
      </BrowserRouter>
    );
    fireEvent.change(await screen.findByRole('select'), {
      target: { value: 'edit' }
    });
    expect(elem).toMatchSnapshot();
  });
  it('Should delete a driver', async () => {
    const handleDeleteMock = jest.fn();
    const elem = render(
      <BrowserRouter>
        <DriverManagement handleDelete={handleDeleteMock} />
      </BrowserRouter>
    );
    fireEvent.change(await screen.findByRole('select'), {
      target: { value: 'delete' }
    });
    const r = new Promise((resolve) => setTimeout(resolve, 1000));
    await r;
    expect(elem).toMatchSnapshot();
  });
  it('Should delete a driver', async () => {
    const handleDeleteMock = jest.fn();
    jest.mock('axios');
    const axiosMock = jest.fn(() => Promise.resolve({}));
    const elem = render(
      <BrowserRouter>
        <DriverManagement handleDelete={handleDeleteMock} axios={axiosMock} />
      </BrowserRouter>
    );
    fireEvent.change(await screen.findByRole('select'), {
      target: { value: 'delete' }
    });
    const r = new Promise((resolve) => setTimeout(resolve, 1000));
    await r;
    expect(elem).toMatchSnapshot();
  });
});
