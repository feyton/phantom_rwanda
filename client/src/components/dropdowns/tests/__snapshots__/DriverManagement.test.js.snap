// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bus Management test Should delete a driver 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10"
      >
        <select
          class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
          id=""
          name=""
          role="select"
        >
          <option
            hidden=""
          >
            Manage
          </option>
          <option
            class="cursor-pointer"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="permission"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10"
    >
      <select
        class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
        id=""
        name=""
        role="select"
      >
        <option
          hidden=""
        >
          Manage
        </option>
        <option
          class="cursor-pointer"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="permission"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Bus Management test Should delete a driver 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10"
      >
        <select
          class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
          id=""
          name=""
          role="select"
        >
          <option
            hidden=""
          >
            Manage
          </option>
          <option
            class="cursor-pointer"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="permission"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10"
    >
      <select
        class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
        id=""
        name=""
        role="select"
      >
        <option
          hidden=""
        >
          Manage
        </option>
        <option
          class="cursor-pointer"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="permission"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Bus Management test Should render drivers management page 1`] = `
<div
  className="z-10"
>
  <select
    className="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
    id=""
    name=""
    onChange={[Function]}
    role="select"
  >
    <option
      hidden={true}
    >
      Manage
    </option>
    <option
      className="cursor-pointer"
      value="edit"
    >
      Edit
    </option>
    <option
      className="cursor-pointer"
      value="delete"
    >
      Delete
    </option>
    <option
      className="cursor-pointer"
      value="permission"
    >
      Permissions
    </option>
  </select>
</div>
`;

exports[`Bus Management test should edit a bus 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10"
      >
        <select
          class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
          id=""
          name=""
          role="select"
        >
          <option
            hidden=""
          >
            Manage
          </option>
          <option
            class="cursor-pointer"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="permission"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10"
    >
      <select
        class="py-1 px-3 font-rale font-bold bg-transparent border rounded-sm"
        id=""
        name=""
        role="select"
      >
        <option
          hidden=""
        >
          Manage
        </option>
        <option
          class="cursor-pointer"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="permission"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
